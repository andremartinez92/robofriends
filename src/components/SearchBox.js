// @flow

import React from 'react';

type Props = {
  onSearchChange: Function,
};

function SearchBox(props: Props) {
  const { onSearchChange } = props;

  return (
    <div className="pa2">
      <input
        className="pa3 ba b--green bg-lightest-blue"
        onChange={onSearchChange}
        placeholder="Search robots"
        type="search"
      />
    </div>
  );
}

export default SearchBox;
