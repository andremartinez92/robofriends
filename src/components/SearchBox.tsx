import React from 'react';

interface Props {
  onSearchChange: (searchField: string) => void,
};

function SearchBox(props: Props) {
  const { onSearchChange } = props;

  return (
    <div className="pa2">
      <input
        aria-label="Search robots"
        className="pa3 ba b--green bg-lightest-blue"
        onChange={event => onSearchChange(event.target.value)}
        placeholder="Search robots"
        type="search"
      />
    </div>
  );
}

export default SearchBox;
