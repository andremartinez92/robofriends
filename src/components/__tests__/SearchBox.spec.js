import React from 'react';
import { shallow } from 'enzyme';

import SearchBox from '../SearchBox';

const onSearchChange = jest.fn();

describe('<SearchBox>', () => {
  it('renders the expected component', () => {
    const subject = shallow(<SearchBox onSearchChange={onSearchChange} />);

    expect(subject).toMatchSnapshot();
  });

  describe('.onSearchChange', () => {
    const subject = shallow(<SearchBox onSearchChange={onSearchChange} />);

    describe('when an event is triggered', () => {
      const event = {
        preventDefault() {
          return null;
        },
        target: { value: 'new search text' },
      };

      it('calls .onSearchChange with the search text', () => {
        subject.find('input').simulate('change', event);

        expect(onSearchChange).toHaveBeenCalledTimes(1);
        expect(onSearchChange).toHaveBeenCalledWith('new search text');
      });
    });
  });
});
