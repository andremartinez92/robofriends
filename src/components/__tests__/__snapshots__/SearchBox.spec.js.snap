// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchBox> renders the expected component 1`] = `
<div
  className="pa2"
>
  <input
    className="pa3 ba b--green bg-lightest-blue"
    onChange={[MockFunction]}
    placeholder="search robots"
    type="search"
  />
</div>
`;
